Simulator/player.py

```diff
@@ -354,13 +354,18 @@ def generate_single_champion_vector(self, curr_champ, champion_info_array):
         item_arr = np.zeros(6)
         for ind, item in enumerate(curr_champ.items):
             if ind <= 2:
-                item_index = list(item_list.keys()).index(item) + 1  # Avoiding index 0 as index 0 is reserved for no items
-                # Hoping to have the observation in terms of components instead but not too sure if the components can be retrieved. Leaving 1 spot empty for now
-                # first spot of each of the item is left empty for now
-                item_arr[ind * 2] = float(item_index) / self.MAX_ITEMS_IN_SET
+                if item in item_builds.keys():
+                    component1, component2 = item_builds[item]
+                    component1_index = list(item_list.keys()).index(component1) + 1  # Avoiding index 0 as index 0 is reserved for no items
+                    component2_index = list(item_list.keys()).index(component2) + 1  # Avoiding index 0 as index 0 is reserved for no items
+                    item_arr[ind] = float(component1_index) / self.MAX_ITEMS_IN_SET

```

Comment:
If you're doing it this way, MAX_ITEMS_IN_SET is going to be like 9. This will also break for items like reforger.